### PR TITLE
Neck and Shoulder Pattern 

### PR DESCRIPTION
**Pull request by DanUOM123**

### CODE REVIEW

File: src/gill/NeckAndShoulderPattern.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        var RShoulderPointZ = get("RShoulderPointZ");
+        var LSideNeckPointZ = get("LSideNeckPointZ");
+        var RSideNeckPointZ = get("RSideNeckPointZ");
+
+        // Build block
+        Block fullBlock = new Block(userName + "_Gill_NeckShoulder");
+        blocks.add(fullBlock);
+
+        //If conditions
+        double Rsidepoint = ((RSideNeckPointZ < 0 &&  RShoulderPointZ < 0) || (RSideNeckPointZ > 0 &&  RShoulderPointZ < 0)) ? abs(RSideNeckPointZ - RShoulderPointZ) : -abs(RSideNeckPointZ - RShoulderPointZ);
```
Comment (aharwood2) [inline src/gill/NeckAndShoulderPattern.java:59]: Can you make your variable naming consistent? The convention elsewhere is the software is to start with a lower case and use camel-case for the reset so these ought to be "sidePointRight". Same for lines 49-52. See line 48 as an example.

File: src/gill/NeckAndShoulderPattern.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        var point2 = new Vector2D(0, -frontNeckDiagonal);
+        var point3 = new Vector2D(-(NeckBaseWidth*0.5), 0);
+        var point4 = new Vector2D(-(NeckBaseWidth*0.5 + shoulderLengthRight), Rsidepoint);
+        var point5 = new Vector2D(NeckBaseWidth*0.5, 0);
+        var point6 = new Vector2D(NeckBaseWidth*0.5 + ShoulderLengthLeft, Lsidepoint);
+
+
+        // All the keypoints/vectors added
+        fullBlock.addKeypoint(point1);
+        fullBlock.addCircularArc(point1, point3, 0.9, true);
```
Comment (aharwood2) [inline src/gill/NeckAndShoulderPattern.java:75]: The multipliers in this section seem a bit arbitrary. Can we add them as Arb_<mesurement_name> variables? See line 40 in the Gill Skirt patterns as an example of arbitrary values.

**Review by aharwood2 — CHANGES_REQUESTED** [review]

File: src/gill/NeckAndShoulderPattern.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Override
+    public void createBlocks()
+    {
+        // Pull from store
+        var shoulderLengthRight = get("shoulderLengthRight");
+        var frontNeckDiagonal = get("frontNeckDiagonal");
+        var backNeckDiagonal = get("backNeckDiagonal");
+
+        // TODO: Daniel to finish parsing measurements into local variables
```
[the remark this answers is not in the document]
Comment (DanUOM123) [reply src/gill/NeckAndShoulderPattern.java]: Todo's removed

File: src/gill/NeckAndShoulderPattern.java (hunk 1 above)
Comment (DanUOM123) [reply src/gill/NeckAndShoulderPattern.java]: variables made more consistent

File: src/gill/NeckAndShoulderPattern.java (hunk 2 above)
Comment (DanUOM123) [reply src/gill/NeckAndShoulderPattern.java:69]: Waiting on ScantoPattern on that

Comment (aharwood2) [reply src/gill/NeckAndShoulderPattern.java:69]: OK let's leave this for now.

**Pull request merged**